### PR TITLE
[LF] cleanning Builtin Function in archive

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: daml-lf/spec/daml-lf-2.rst [hunk 1]
```diff
@@ -4116,83 +4106,52 @@ Numeric functions
...
+* ``MUL_NUMERIC : ∀ (α₁ α₂ α : nat) . 'Numeric' α → 'Numeric' α₁ → 'Numeric' α₂ → 'Numeric' α``
 
   Multiplies the two numerics and rounds the result to the closest
   multiple of ``10⁻ᵅ`` using `banker's rounding convention
   <https://en.wikipedia.org/wiki/Rounding#Round_half_to_even>`_.  The
   type parameters `α₁`, `α₂`, `α` define the scale of the first input,
-  the second input, and the output, respectively.  Throws an
-  ``ArithmeticError`` exception in case of overflow.
+  the second input, and the output, respectively.  The first argument
+  is used as a witness for the scale `α`, is actual value is ignored.
```
Comment (paulbrauner-da) [inline daml-lf/spec/daml-lf-2.rst:4116]:
```suggestion
  is used as a witness for the scale `α`, its actual value is ignored.
```

Comment (paulbrauner-da) [reply daml-lf/spec/daml-lf-2.rst:4116]: Is its value actually ignored? I thought the value was providing the scale.

File: daml-lf/spec/daml-lf-2.rst [hunk 2]
```diff
@@ -4116,83 +4106,52 @@ Numeric functions
...
-* ``DIV_NUMERIC_LEGACY : ∀ (α₁ α₂ α : nat) . 'Numeric' α₁ → 'Numeric' α₂ → 'Numeric' α``
+* ``DIV_NUMERIC : ∀ (α₁ α₂ α : nat) . 'Numeric' α → 'Numeric' α₁ → 'Numeric' α₂ → 'Numeric' α``
 
   Divides the first decimal by the second one and rounds the result to
   the closest multiple of ``10⁻ᵅ`` using `banker's rounding convention
   <https://en.wikipedia.org/wiki/Rounding#Round_half_to_even>`_ (where
   `n` is given as the type parameter).  The type parameters `α₁`,
   `α₂`, `α` define the scale of the first input, the second input, and
-  the output, respectively.  Throws an ``ArithmeticError`` exception
+  the output, respectively.  The first argument is used as a witness for the scale `α`, is actual value is ignored. Throws an ``ArithmeticError`` exception
```
Comment (paulbrauner-da) [inline daml-lf/spec/daml-lf-2.rst:4127]:
```suggestion
  the output, respectively.  The first argument is used as a witness for the scale `α`, its actual value is ignored. Throws an ``ArithmeticError`` exception
```

File: daml-lf/spec/daml-lf-2.rst [hunk 3]
```diff
@@ -4116,83 +4106,52 @@ Numeric functions
...
   if the second argument is ``0.0`` or if the computation overflow.
 
-* ``CAST_NUMERIC_LEGACY : ∀ (α₁, α₂: nat) . 'Numeric' α₁ → 'Numeric' α₂``
+* ``CAST_NUMERIC : ∀ (α₁, α₂: nat) . 'Numeric' α₂ →  'Numeric' α₁ → 'Numeric' α₂``
 
   Converts a decimal of scale `α₁` to a decimal scale `α₂` while
-  keeping the value the same. Throws an ``ArithmeticError`` exception
+  keeping the value the same. The first argument is used as a witness for the scale `α`, is actual value is ignored. Throws an ``ArithmeticError`` exception
```
Comment (paulbrauner-da) [inline daml-lf/spec/daml-lf-2.rst:4133]:
```suggestion
  keeping the value the same. The first argument is used as a witness for the scale `α`, its actual value is ignored. Throws an ``ArithmeticError`` exception
```

File: daml-lf/spec/daml-lf-2.rst [hunk 4]
```diff
@@ -4116,83 +4106,52 @@ Numeric functions
...
-  the type parameter `α`.
-
-* ``EQUAL_NUMERIC : ∀ (α : nat) . 'Numeric' α → 'Numeric' α → 'Bool'``
-
-  Returns ``'True'`` if the first numeric is equal to the second,
-  ``'False'`` otherwise.  The scale of the inputs is given by the type
-  parameter `α`.
-
-  [*Available in version < 1.11*]
+  multiplied by `1E(α₁-α₂)`. The first argument is used as a witness for the scale `α`, is actual value is ignored.
```
Comment (paulbrauner-da) [inline daml-lf/spec/daml-lf-2.rst:4140]:
```suggestion
  multiplied by `1E(α₁-α₂)`. The first argument is used as a witness for the scale `α`, its actual value is ignored.
```

File: daml-lf/spec/daml-lf-2.rst [hunk 5]
```diff
@@ -4116,83 +4106,52 @@ Numeric functions
...
+* ``TEXT_TO_NUMERIC : ∀ (α : nat) . 'Numeric' α →'Text' → 'Optional' 'Numeric' α``
 
   Given a string representation of a numeric returns the numeric
   wrapped in ``Some``. If the input does not match the regexp
   ``[+-]?\d+(\.d+)?`` or if the result of the conversion cannot
   be mapped into a decimal without loss of precision, returns
-  ``None``.  The scale of the output is given by the type parameter
-  `α`.
+  ``None``.  The scale of the output is given by the  parameter
+  `α`. The first argument is used as a witness for the scale `α`, is actual value is ignored.
```
Comment (paulbrauner-da) [inline daml-lf/spec/daml-lf-2.rst:4154]:
```suggestion
  `α`. The first argument is used as a witness for the scale `α`, its actual value is ignored.
```

File: daml-lf/spec/daml-lf-2.rst [hunk 6]
```diff
@@ -4202,21 +4161,21 @@ BigNumeric functions
   Adds the two decimals. Throws an ``ArithmeticError`` if the output
   is not a valid BigNumeric.
 
-  [*Available in version ≥ 1.13*]
+  [*Available in version ≥ 1.dev*]
```
Comment (paulbrauner-da) [inline daml-lf/spec/daml-lf-2.rst:4164]: Shouldn't that be 2.1? (Here and below.)

File: daml-lf/spec/daml-lf-2.rst [hunk 7]
```diff
@@ -4252,49 +4211,49 @@ BigNumeric functions
...
-  [*Available in version ≥ 1.13*]
+  [*Available in version ≥ 2.dev*]
 
-* ``'BIGNUMERIC_TO_NUMERIC_LEGACY' : ∀ (α : nat). 'BigNumeric'  → 'Numeric' α``
+* ``'BIGNUMERIC_TO_NUMERIC' : ∀ (α : nat). 'Numeric' α` → 'BigNumeric' → 'Numeric' α``
 
   Converts the ``BigNumeric`` to a ``Numeric α`` value with scale
   ``α``.  Throws an ``ArithmeticError`` in case the result cannot be
-  represented without loss of precision.
+  represented without loss of precision. The first argument is used as a witness for the scale `α`, is actual value is ignored.
```
Comment (paulbrauner-da) [inline daml-lf/spec/daml-lf-2.rst:4248]:
```suggestion
  represented without loss of precision. The first argument is used as a witness for the scale `α`, its actual value is ignored.
```

File: daml-lf/spec/daml-lf-2.rst [hunk 8]
```diff
@@ -669,7 +669,7 @@ Then we can define our kinds, types, and expressions::
        |  LitTimestamp                              -- ExpLitTimestamp: UTC timestamp literal
        |  LitParty                                  -- ExpLitParty: Party literal
        |  cid                                       -- ExpLitContractId: Contract identifiers
-       |  LitRoundingMode                           -- ExpLitRoundingMode: Rounding Mode
+       |  LitRoundingMode                           -- ExpLitRoundingMode: Rounding Mode [Daml-LF ≥ 1.dev]
```
Comment (paulbrauner-da) [inline daml-lf/spec/daml-lf-2.rst:672]: Should it be 2.1?

**Review by paulbrauner-da — APPROVED** [review]
Nice cleanup!

File: daml-lf/spec/daml-lf-2.rst (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/spec/daml-lf-2.rst:4116]:
```suggestion
  is used as a witness for the scale `α`.
```

File: daml-lf/spec/daml-lf-2.rst (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/spec/daml-lf-2.rst:4127]:
```suggestion
  the output, respectively.  The first argument is used as a witness for the scale `α`. Throws an ``ArithmeticError`` exception
```

File: daml-lf/spec/daml-lf-2.rst (hunk 3 above)
Comment (remyhaemmerle-da) [reply daml-lf/spec/daml-lf-2.rst:4133]:
```suggestion
  keeping the value the same. The first argument is used as a witness for the scale `α`. Throws an ``ArithmeticError`` exception
```

File: daml-lf/spec/daml-lf-2.rst (hunk 4 above)
Comment (remyhaemmerle-da) [reply daml-lf/spec/daml-lf-2.rst:4140]:
```suggestion
  multiplied by `1E(α₁-α₂)`. The first argument is used as a witness for the scale `α`.
```

File: daml-lf/spec/daml-lf-2.rst (hunk 5 above)
Comment (remyhaemmerle-da) [reply daml-lf/spec/daml-lf-2.rst:4154]:
```suggestion
  `α`. The first argument is used as a witness for the scale `α`.
```

File: daml-lf/spec/daml-lf-2.rst (hunk 6 above)
Comment (remyhaemmerle-da) [inline daml-lf/spec/daml-lf-2.rst:4164]:
```suggestion
  [*Available in version ≥ 2.dev*]
```

Comment (remyhaemmerle-da) [reply daml-lf/spec/daml-lf-2.rst:4164]: no 2.dev

File: daml-lf/spec/daml-lf-2.rst [hunk 9]
```diff
@@ -4202,21 +4161,21 @@ BigNumeric functions
...
-  [*Available in version ≥ 1.13*]
+  [*Available in version ≥ 1.dev*]
 
 * ``SUB_BIGNUMERIC : 'BigNumeric' → 'BigNumeric' → 'BigNumeric'``
 
   Subtracts the two decimals. Throws an ``ArithmeticError`` if the
   output is not a valid BigNumeric.
 
-  [*Available in version ≥ 1.13*]
+  [*Available in version ≥ 1.dev*]
```
Comment (remyhaemmerle-da) [inline daml-lf/spec/daml-lf-2.rst:4171]:
```suggestion
  [*Available in version ≥ 2.dev*]
```

File: daml-lf/spec/daml-lf-2.rst [hunk 10]
```diff
@@ -4202,21 +4161,21 @@ BigNumeric functions
...
 
 * ``MUL_BIGNUMERIC : 'BigNumeric' → 'BigNumeric' → 'BigNumeric'``
 
   Multiplies the two numerics. Throws an ``ArithmeticError`` if the
   output is not a valid BigNumeric.
 
-  [*Available in version ≥ 1.13*]
+  [*Available in version ≥ 1.dev*]
```
Comment (remyhaemmerle-da) [inline daml-lf/spec/daml-lf-2.rst:4178]:
```suggestion
  [*Available in version ≥ 2.dev*]
```

**Review by akrmn — APPROVED** [review]
good riddance! thank you @remyhaemmerle-da

**Pull request merged**